hazelcast-client-legacy/src/test/java/com/hazelcast/client/replicatedmap/ClientReplicatedMapListenerTest.java

```diff
@@ -115,6 +117,50 @@ public void run() throws Exception {
         });
     }
 
+
+    @Test
+    public void testListenToKeyForEntryAdded() throws Exception {
+        HazelcastInstance instance = factory.newHazelcastInstance();
+        HazelcastInstance client = factory.newHazelcastClient();
+        String mapName = randomMapName();
+        ReplicatedMap<Object, Object> replicatedMap = client.getReplicatedMap(mapName);
+        final EventCountingListener listener = new EventCountingListener();
+        replicatedMap.addEntryListener(listener, 1);
+        replicatedMap.put(1, 1);
+        replicatedMap.put(2, 2);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                Assert.assertEquals(1, listener.addCount.get());
```

Comment:
If the event would not contain the right content, your count based test is not going to detect this; perhaps 2 was received instead of 1. If you would store the events in a threadsafe list in the entry listener, you could do assertEquals(asList(1), listener.items())

There is another testing problem that can be hidden. Imagine that the predicate would not be triggered correctly (ignored) and therefor the listener received 1 first and then it received 2. 

But you end the test immediately after you have found the first event.. you are not going to check for some time if the second event is not going to be received.